src/js/memory/solverMemory.ts

```diff
@@ -0,0 +1,270 @@
+import { getById } from '../utilities';
+
+const position = "Position";
+const label = "Label";
+
+interface ElementData {
+	name: string;
+	stage: number;
+	value: number;
+  }
+
+export function memory_solve(e : HTMLInputElement): void {		
+	clean_form();
+
+	//only solve if checking a button
+	if (!e.checked) {
+		return;
+	}
+
+	var n = e.getAttribute("data-elementname");
+	var s =  e.getAttribute("data-elementstage");
+	var  v = e.getAttribute("data-elementvalue");
+
+	if (n && s && v) {
+		const element: ElementData = {
+			name: n,
+			stage: parseInt(s),
+			value: parseInt(v),
+		  };
```

Comment:
..but you could eliminate the short-lived variables like this
```suggestion
  const element: ElementData = {
	  name: e.getAttribute("data-elementname"),
	  stage: parseInt(e.getAttribute("data-elementstage")),
	  value: parseInt(e.getAttribute("data-elementvalue")),
  };
  if (ElementData.name && ElementData.stage && ElementData.value) {
```
although I'm not sure if the interface would have to explicitly define those as nullable.